cmd/internal/server/core.go

```diff
@@ -203,16 +204,24 @@ func (s *Server) Stop() {
 
 func (s *Server) loadPlugins(registry *plugin.Registry) {
 	log.Debug("Loading plugins")
+	var wg sync.WaitGroup
+	var fail sync.Once
 	anyFailed := false
+
 	for name, root := range s.plugins {
 		log.Infof("Loading %v", name)
-		if err := registry.RegisterPlugin(root, s.opts.PluginConfig[name]); err != nil {
-			// %+v is a convention used by some errors to print additional context such as a stack trace
-			log.Warnf("%v failed to load: %+v", name, err)
-			anyFailed = true
-		}
+		wg.Add(1)
+		go func(name string, root plugin.Root) {
+			if err := registry.RegisterPlugin(root, s.opts.PluginConfig[name]); err != nil {
```

Comment:
Do we need a lock in https://github.com/puppetlabs/wash/blob/master/plugin/registry.go#L38 ?

Reply:
Yes, we probably do. Good catch.